internal/cache/file/cache_handle_test.go

```diff
@@ -16,307 +16,385 @@ package file
 
 import (
 	"context"
+	"crypto/rand"
+	"errors"
 	"os"
+	"path"
+	"strings"
 	"testing"
 
 	"github.com/googlecloudplatform/gcsfuse/internal/cache/data"
 	"github.com/googlecloudplatform/gcsfuse/internal/cache/file/downloader"
 	"github.com/googlecloudplatform/gcsfuse/internal/cache/lru"
 	"github.com/googlecloudplatform/gcsfuse/internal/cache/util"
+	"github.com/googlecloudplatform/gcsfuse/internal/locker"
+	"github.com/googlecloudplatform/gcsfuse/internal/storage"
 	"github.com/googlecloudplatform/gcsfuse/internal/storage/gcs"
+	"github.com/googlecloudplatform/gcsfuse/internal/storage/storageutil"
+	"github.com/googlecloudplatform/gcsfuse/tools/integration_tests/util/operations"
 	. "github.com/jacobsa/ogletest"
 )
 
-const CacheMaxSize = 50
-const TestFilePath = "test_file.txt"
-const DefaultFileMode = 0644
-const TestFileContent = "abcdefghijklmnop"
-const DstBufferLen = 5
-const TestOffset = 5
+const CacheMaxSize = 100 * util.MiB
+const ReadContentSize = 1 * util.MiB
 
-const FileContentFromTestOffset = "fghij"
-
-const TestBucketName = "test-bucket"
-const TestObjectName = "test.txt"
-const TestObjectNameNotInFileInfoCache = "test_not_in_file_info.txt"
-const TestObjectGeneration = 3434343
-const TestFileInfoFileSize = 32
+const TestObjectSize = 20 * util.MiB
+const TestObjectName = "foo.txt"
+const DefaultSequentialReadSizeMb = 16
 
 func TestCacheHandle(t *testing.T) { RunTests(t) }
 
 type cacheHandleTest struct {
-	ch *CacheHandle
+	bucket        gcs.Bucket
+	fakeStorage   storage.FakeStorage
+	object        *gcs.MinObject
+	cache         *lru.Cache
+	cacheHandle   *CacheHandle
+	cacheLocation string
+	fileSpec      data.FileSpec
 }
 
-// Mocking test bucket, which contains Name() method.
-type testBucket struct {
-	gcs.Bucket
-	BucketName string
+func init() {
+	RegisterTestSuite(&cacheHandleTest{})
 }
 
-func (tb testBucket) Name() string {
-	return tb.BucketName
+func (cht *cacheHandleTest) addTestFileInfoEntryInCache() {
+	// Add an entry into
+	fileInfoKey := data.FileInfoKey{
+		BucketName: storage.TestBucketName,
+		ObjectName: TestObjectName,
+	}
+	fileInfo := data.FileInfo{
+		Key:              fileInfoKey,
+		ObjectGeneration: cht.object.Generation,
+		FileSize:         cht.object.Size,
+		Offset:           0,
+	}
+	fileInfoKeyName, err := fileInfoKey.Key()
+	AssertEq(nil, err)
+
+	_, err = cht.cache.Insert(fileInfoKeyName, fileInfo)
+	AssertEq(nil, err)
 }
 
-// Test helper.
-func createFileWithContent(filePath string, content string) (*os.File, error) {
-	fileSpec := data.FileSpec{
-		Path: filePath,
-		Perm: os.FileMode(DefaultFileMode),
-	}
+// Todo (princer) - enable this after the fix in job.go.
```

Comment:
as discussed, this is shortcoming on fake storage, so add link to https://github.com/GoogleCloudPlatform/gcsfuse/blob/read_cache_release/internal/cache/file/downloader/job_test.go#L324

Also, you can add unit tests where job actually downloads the data by keep sequentialReadSizeMb greater than equal to content size.